pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -781,7 +781,7 @@ func makeSwapIndexSpec(
 func fallBackIfZoneConfigExists(b BuildCtx, n tree.NodeFormatter, id catid.DescID) {
 	{
 		tableElts := b.QueryByID(id)
-		if _, _, elem := scpb.FindTableZoneConfig(tableElts); elem != nil {
+		if _, _, elem := scpb.FindIndexZoneConfig(tableElts); elem != nil {
 			panic(scerrors.NotImplementedErrorf(n,
 				"regional by row partitioning is not supported"))
```

Comment:
My main question is that why don't we just check the `TableLocalityRegionalByRow` element?

Reply:
Good point, this one needs the comment updated and a few users only need the regional by-row checks. Let me adjust this